Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/SqlDialectTest.cs

```diff
@@ -49,5 +57,219 @@ public void GetParameterName ()
+      var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+      var storageTypeInformation = new StorageTypeInformation(
+          typeof(char[]),
+          "test",
+          DbType.AnsiString,
+          false,
+          5,
+          typeof(char[]),
+          _typeConverterStub.Object);
+
+      var convertedValue = new char[10];
+      _typeConverterStub.Setup(_ => _.ConvertTo(null, null, "value", storageTypeInformation.StorageType)).Returns(convertedValue);
+
+      commandMock.Setup(_ => _.CreateParameter()).Returns(dataParameterMock.Object).Verifiable();
+
+      dataParameterMock.SetupSet(_ => _.ParameterName = "@dummy").Verifiable();
+      dataParameterMock.SetupSet(_ => _.DbType = storageTypeInformation.StorageDbType).Verifiable();
+      dataParameterMock.SetupSet(_ => _.Value = convertedValue).Verifiable();
+
+      var result = _dialect.CreateDataParameter(commandMock.Object, storageTypeInformation, "@dummy","value");
+
+      commandMock.Verify();
+      dataParameterMock.Verify();
+
+      Assert.That(result, Is.SameAs(dataParameterMock.Object));
+      dataParameterMock.VerifySet(_ => _.Size = It.IsAny<int>(), Times.Never());
+    }
+
+    [Test]
+    public void CreateDataParameter_WithByteArrayValueExceedingFixedSize_DoesNotInitializeSize ()
+    {
+      var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+      var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+      var storageTypeInformation = new StorageTypeInformation(
+          typeof(byte[]),
+          "test",
+          DbType.Binary,
+          false,
+          5,
+          typeof(byte[]),
+          _typeConverterStub.Object);
+
+      var convertedValue = new byte[10];
+      _typeConverterStub.Setup(_ => _.ConvertTo(null, null, "value", storageTypeInformation.StorageType)).Returns(convertedValue);
+
+      commandMock.Setup(_ => _.CreateParameter()).Returns(dataParameterMock.Object).Verifiable();
+
+      dataParameterMock.SetupSet(_ => _.ParameterName = "@dummy").Verifiable();
+      dataParameterMock.SetupSet(_ => _.DbType = storageTypeInformation.StorageDbType).Verifiable();
+      dataParameterMock.SetupSet(_ => _.Value = convertedValue).Verifiable();
+
+      var result = _dialect.CreateDataParameter(commandMock.Object, storageTypeInformation, "@dummy","value");
+
+      commandMock.Verify();
+      dataParameterMock.Verify();
+
+      Assert.That(result, Is.SameAs(dataParameterMock.Object));
+      dataParameterMock.VerifySet(_ => _.Size = It.IsAny<int>(), Times.Never());
+    }
+
+    [Test]
+    public void CreateDataParameter_WithNullResult ()
+    {
+      var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+      var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+      _typeConverterStub.Setup(_ => _.ConvertTo(null, null, "value", _storageTypeInformation.StorageType)).Returns((object)null);
```

Comment:
here, for isntance, it's actually difficult to see that the parameter has no size. you need to scroll up to the setup to understand this.